test/it/uk/gov/hmrc/apinotificationpull/acceptance/RetrieveAndDeleteNotificationSpec.scala

```diff
@@ -18,58 +18,153 @@ package uk.gov.hmrc.apinotificationpull.acceptance
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration.wireMockConfig
 import org.scalatest.OptionValues._
+import org.scalatest._
 import org.scalatestplus.play.guice.GuiceOneAppPerTest
+import play.api.Application
+import play.api.inject.guice.GuiceApplicationBuilder
 import play.api.test.FakeRequest
 import play.api.test.Helpers._
 
-class RetrieveAndDeleteNotificationSpec extends FeatureSpec with GivenWhenThen with Matchers with GuiceOneAppPerTest {
+class RetrieveAndDeleteNotificationSpec extends FeatureSpec with GivenWhenThen with Matchers with GuiceOneAppPerTest
+  with BeforeAndAfterEach with BeforeAndAfterAll {
 
-  feature("Retrieve a single message from the Pull Service (DELETE)") {
+  private val clientId = "client-id"
+  private val xClientIdHeader = "X-Client-ID"
+
+  private val externalServicesHost = "localhost"
+  private val externalServicesPort = 11111
+
+  override def newAppForTest(testData: TestData): Application = new GuiceApplicationBuilder().configure(Map(
+    "microservice.services.api-notification-queue.host" -> externalServicesHost,
+    "microservice.services.api-notification-queue.port" -> externalServicesPort
+  )).build()
+
+  val externalServices: WireMockServer = new WireMockServer(wireMockConfig.port(externalServicesPort))
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    if(!externalServices.isRunning) externalServices.start()
+    WireMock.configureFor(externalServicesHost, externalServicesPort)
+  }
+
+  override def afterAll(): Unit = {
+    super.afterAll()
+    externalServices.stop()
+  }
+
+  feature("GET all notifications from the API Notification Pull service") {
+    info("As a 3rd Party")
+    info("I want to successfully retrieve all notification locations by client id")
+
+    val validRequest = FakeRequest("GET", "/").
+      withHeaders(ACCEPT -> "application/vnd.hmrc.1.0+xml", xClientIdHeader -> clientId)
+
+    scenario("Successful GET and 3rd party receives the notifications locations") {
+      // TODO
+    }
+
+    scenario("Missing Accept Header") {
+      Given("You do not provide the Accept Header")
+      val request = validRequest.copyFakeRequest(headers = validRequest.headers.remove(ACCEPT))
+
+      When("You call make the 'GET' call to the api-notification-pull service")
+      val result = route(app = app, request).value
+
+      Then("You will be returned a 406 error response")
+      status(result) shouldBe NOT_ACCEPTABLE
+      contentAsString(result) shouldBe ""
+    }
+
+    scenario("Missing X-Client-Id Header") {
+      Given("The platform does not inject a X-Client-Id Header")
+      val request = validRequest.copyFakeRequest(headers = validRequest.headers.remove(xClientIdHeader))
+
+      When("You call make the 'GET' call to the api-notification-pull service ")
+      val result = route(app = app, request).value
+
+      Then("You will be returned a 500 error response")
+      status(result) shouldBe INTERNAL_SERVER_ERROR
```

Comment:
I think we should change the HeaderValidator rules.
`400 Bad Request` is a better error code to return when the client id header is missing, since the request is invalid.
500 is usually used for internal system errors.

Reply:
I've confirmed with @crystalmark that we should be returning 500 for missing X-Client-ID header as it will be added by the platform, not the external client